packages/vite-plugin-cloudflare/src/miniflare-options.ts

```diff
@@ -216,7 +216,18 @@ export function getMiniflareOptions(
 						if (moduleId.startsWith('cloudflare:')) {
 							const result = {
 								externalize: moduleId,
-								type: 'module',
+								type: 'builtin',
+							} satisfies vite.FetchResult;
+
+							return new MiniflareResponse(JSON.stringify(result));
+						}
+
+						// Sometimes Vite fails to resolve built-ins and converts them to "url-friendly" ids
+						// that start with `/@id/...`.
```

Comment:
This is where the wrapping/unwrapping code is : 

https://github.com/vitejs/vite/blob/f0f665941352b377de01ba709029d5e8b81c084c/packages/vite/src/shared/utils.ts#L6-L24

and here is the place where I think the wrapping actually happens in this case:

https://github.com/vitejs/vite/blob/f0f665941352b377de01ba709029d5e8b81c084c/packages/vite/src/node/plugins/importAnalysis.ts#L132-L138

Reply:
I think, from looking at the code, that it is safe to unwrap module IDs anywhere (it happens a lot on the Vite code).

I don't think this is a public API.
I actually think this might be a bug in Vite but I couldn't work out where.

I was kind of hoping that a new version update to Vite might fix this...

Reply:
Sorry, when I say built-ins, I actually mean Node.js built-ins, which include things like `path`.

So sure, we could embed this list into the function, but that seems overkill.
Moreover, imports (in our world) that start with `/@id/...` are never going to be valid imports as-is. So I don't worry that this can break normal user code.

Reply:
Curious why this came up in this PR given that this code doesn't execute during build. Did adding the additional externals to `rollupOptions` lead to `/@id/` being added here?

Reply:
This came up because the only way to fix the builds was to remove the filtering of aliases that don't point to unenv from the aliases passed to Vite.

But adding back those aliases broke the dev behaviour. It's a catch-22.

I spent some time trying to work out a more clean solution and this is the only one I could come up with. If you want to weave some of your magic and fix this more correctly, I would be very happy @jamesopstad.

As I mention above, I suspect that this is a bug in Vite, but I also couldn't work out where. If we could identify that bug, we could propose a fix to Vite.

Do we want to block landing this on such investigation?

Reply:
No, I don't think this should block us for now. Let's land this and then investigate again once we have CI for builds in place.

Reply:
Thanks for the explanation by the way. Makes sense. I've added an issue so that we can come back to this: https://github.com/flarelabs-net/vite-plugin-cloudflare/issues/63